### PR TITLE
Toolbar tool updates

### PR DESCRIPTION
**Pull request by sashadev-sky**
References #246, https://github.com/publiclab/mapknitter/issues/124  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updates
* [x] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

=====================================================

Updates

1) Keymapper order re-arranged to match toolbar order

2) Update `RotateScale` -> `Rotate+Scale` in hovertext and keymapping and fix hovertext toggle

3) Remove deprecated `charset` attribute from our `link` tags in `index.html`  and `select.html` and unnecessary `type="text/css"` (`rel=stylesheet` is sufficient)

4) Switched to material design icons :
   - Offers much more flexibility for customizing and has every icon in solid, outlined, two-toned, rounded, and sharp for free. Font awesome only lets you have their least appealing icons for free
   - Has way more options. Font awesome does not even have rotation icons in their latest release (which would have given us vector scale): https://fontawesome.com/icons?d=gallery&q=rotate so 
 
5) Fixed `_toggleOutline` method: fixed bug where we were also toggling image opacity here not just the outline opacity.

6) `_toggleOrder`

- we have a client side drag and drop now in mapknitter, and in LDI I don't think it's ever going to make sense to try to specifically target images z-index with these methods. I think it's a good fix for now to just send it to the top or send it to the back and it works (if you want an image stacked in the middle, send it up above one and send the next up above it). 

   - **pending confirmation**: I can open a new issue to create a cool drag and drop minimap like view?

- Fixed hotkeys:  (these was probably a plan in place to fix sendUp and sendDown to make them more complex but they are the same as _ToggleOrder right now except they don't swap the `this._toggledImage` property. I use that property to make the tool icon dynamic so changed it back. (also again don't think it'll ever make sense to use a hotkey to target specific z-indexes)

```   
'j': "_sendUp",        -> "_toggleOrder"
'k': "_sendDown",  ->  "_toggleOrder"
```

7) bumped leaflet from  `"leaflet": "^1.0.0"` to `"leaflet": "^1.5.1"` in `package.json` devDependencies. 

=====================================================

=====================================================

**pending**
- name of the action in the `mode` to also be updated to use `rotate+scale`?

=====================================================

### CODE REVIEW

**Comment by sashadev-sky** [comment]
@jywarren @rexagod The hovertext is fixed:
hovertext

**for the icons** need your input! 

I realized the reason they are confusing is because the current distort icon is an icon that typically represents "rotate". I would say: 

step 1: maybe we should swap them 
step 2: replace the icon that represents an image for this https://fontawesome.com/icons/vector-square?style=solid

let me know your thoughts!

**Comment by jywarren** [comment]
This looks super. Shall we call it "Rotate/Scale" instead of "RotateScale"?

**Comment by jywarren** [comment]
And vectorscale icon is awesome!

**Comment by sashadev-sky** [comment]
@jywarren I updated `RotateScale` to `Rotate+Scale` in the hovertext and keymapper because it seemed to make the most sense aesthetically, since we might use a slash for something like "Send up / down". What do you think?

Also did you want the name of the action in the `mode` to also be updated to use `rotate+scale`? It is currently `rotateScale`? This would be the name change for the option the user can pass in to set a default tool. 

@rexagod

**Comment by sashadev-sky** [comment]
@jywarren @rexagod please give this a final review! Lots of changes (see PR description). Here is the UI now:
tools

**Comment by rexagod** [comment]
Okay wow! This looks superb! Reviewing this ASAP.

File: src/edit/DistortableImage.EditToolbar.js [hunk 1]
```diff
@@ -1,182 +1,224 @@
...
+      editing._toggleOrder();
+      this.disable();
+    }
+	}),
+	
+  EnableEXIF = EditOverlayAction.extend({
+    options: {
+      toolbarIcon: {
+        html: '<i class="material-icons-outlined md-18">explore</i>',
+        tooltip: 'Geocode Image'
```
Comment (rexagod) [inline src/edit/DistortableImage.EditToolbar.js]: Just a minor doubt, should this be "geolocate image" instead, or do you think it's good as is?

Comment (sashadev-sky) [reply src/edit/DistortableImage.EditToolbar.js]: done!

**Comment by sashadev-sky** [comment]
@jywarren would we be ok to merge this? I will follow up with a PR for documentation updates

**Comment by jywarren** [comment]
Love this! However, let's get input from Sairam and Mo -- @stefannibrasil where is the best place to ask their input on this icon change?

But yes, otherwise I am very +1 to merging!

**Comment by sashadev-sky** [comment]
> Love this! However, let's get input from Sairam and Mo -- @stefannibrasil where is the best place to ask their input on this icon change?
> 
> But yes, otherwise I am very +1 to merging!

Sounds good! I have not met Sairam and Mo yet are they heading UI updates across PL or mapknitter?

Also - i'll do this in a follow up PR if yes but did you want `rotateScale` to be changed to `rotate+scale` as a `mode` too? see https://github.com/publiclab/Leaflet.DistortableImage/pull/255#issuecomment-497992896

**Comment by jywarren** [comment]
They're providing input on UI work, because they are running mapping events
and using Mapknitter! Let me find their usernames on publiclab

**Comment by jywarren** [comment]
Ah! They've been posting at https://publiclab.org/tag/community-atlas

**Comment by sashadev-sky** [comment]
@jywarren ah ok! I see their usernames! If @stefannibrasil doesn't get back to us I can post a question directed at them and tag them??

**Comment by jywarren** [comment]
For sure, actually you can go ahead now if you like!

**Comment by sashadev-sky** [comment]
@jywarren @rexagod Mo said he liked the update so I will merge now :)
https://publiclab.org/questions/molangmuir10/05-30-2019/how-can-i-download-pi-builder-sd-card-image-with-the-wifi-network-for-camera-access-infragram-and-image-sequencer#answer-0-comment

**Pull request merged**